modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/footprint/MultiLevelROIProviderFactory.java

```diff
@@ -0,0 +1,129 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2015 - 2016, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.coverage.grid.io.footprint;
+
+import java.io.File;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.geotools.data.DataUtilities;
+import org.geotools.resources.coverage.CoverageUtilities;
+import org.geotools.util.Utilities;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+/**
+ * Factory class used for returning a {@link MultiLevelROIProvider} based on the input footprint properties
+ * and files
+ * 
+ * @author Andrea Aime, GeoSolutions
+ * @author Nicola Lagomarsini, GeoSolutions
+ * @author Daniele Romagnoli, GeoSolutions
+ */
+public class MultiLevelROIProviderFactory {
+
+    private final static Logger LOGGER = org.geotools.util.logging.Logging
+                    .getLogger(MultiLevelROIProviderFactory.class.toString());
+
+    // well known properties
+    public static final String SOURCE_PROPERTY = "footprint_source";
+
+    // store types
+    public static final String TYPE_SIDECAR = "sidecar";
+    public static final String TYPE_RASTER = "raster";
+
+    protected MultiLevelROIProviderFactory() {
+    }
+
+    /**
+     * Builds a footprint provider from file location
+     * 
+     * @param the reference file. 
+     * @return
+     * @throws Exception
+     * 
+     * @see {@link #createFootprintProvider(File, Geometry)}
+     */
+    public static MultiLevelROIProvider createFootprintProvider(File referenceFile) {
+        return createFootprintProvider(referenceFile, null);
+    }
+
+    /**
+     * Builds a footprint provider from file location
+     * 
+     * @param It can be: 
+     *      - a folder containing the footprint config files (if any) OR the footprint itself. 
+     *      - the data file for which we are looking for a footprint.
+     * @param The optional granuleBounds geometry. if not null, it will be used as data's reference geometry.
+     * @return
+     * @throws Exception
+     */
+    public static MultiLevelROIProvider createFootprintProvider(final File referenceFile, Geometry granuleBounds) {
+        Utilities.ensureNonNull("referenceFile", referenceFile);
+        File configDirectory = referenceFile.isDirectory() ? referenceFile : referenceFile.getParentFile();
+        File configFile = new File(configDirectory, "footprints.properties");
```

Comment:
So if several raster files that are not a mosaic are in the same directory, they will all use the same configuration file for footprints? Might make sense, it's unlikely that one has several different footprint policies, but it might make sense to look also for a sidecar <refFile>-footprints.properties so that one can customize behavior per file (e.g., insets for a specific file could have to be different)